### PR TITLE
Ncnews 5

### PR DESCRIPTION
**Pull request by thendersondev**
Ticket 5
- Add comment_count functionality to endpoint GET /api/articles/:article_id
- refactor some tests to remove use of convertTimestampToDate
- EDIT: refactor to make comment_count a key on the article object, instead of being a key on the body object
- EDIT2: refactor as per popatre comments, using psql join to solve ticket

### CODE REVIEW

File: controllers/articles-controllers.js [hunk 1]
```diff
@@ -17,7 +18,11 @@ exports.getArticles = async (req, res, next) => {
 exports.getArticleById = async (req, res, next) => {
   try {
     const { article_id: id } = req.params;
-    const article = await fetchArticle(id);
+    const [article, { rows: comments }] = await Promise.all([
+      fetchArticle(id),
+      fetchCommentsByArticleId(id),
```
Comment (popatre) [inline controllers/articles-controllers.js]: The idea is that you query the database using a JOIN to comments. 

The foreign key should line up in comments to article id, then you can use aggregate functions to count the comments. 
At the moment, you're bypassing a lot of the PQSL functionality and the point of having a relational database.

File: models/comments-models.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const db = require("../db/connection");
+
+exports.fetchCommentsByArticleId = async (id) => {
+  return await db.query(`SELECT * FROM comments WHERE article_id = $1`, [id]);
```
Comment (popatre) [inline models/comments-models.js]: This functionality should be covered in the join

File: controllers/articles-controllers.js [hunk 3]
```diff
@@ -17,7 +18,11 @@ exports.getArticles = async (req, res, next) => {
...
+    ]);
+    article.comment_count = comments.length;
```
Comment (popatre) [inline controllers/articles-controllers.js]: This functionality should be covered in the aggregate function 'COUNT'

**Review by popatre — CHANGES_REQUESTED** [review]

**Review by DominicFH — APPROVED** [review]
Nice stuff 👍

**Pull request merged**